File: modulo 6/credit.py
import sys

tam = 0
num = 0
num1=0
num2=0
dig1 = 0
dig2 = 0

num = input("INSIRA O NUMERO DO CARTÃO: ")
tam = len(str(num))

if(tam < 13 or tam > 16):
    print("INVALID")
    sys.exit()

for i in range(-2, -(tam+1), -2):
    num1 = int(num[i])*2
    dig1 = str(dig1)+str(num1)


for y in range(-1, -(tam+1),-2):
    dig2 = int(num[y])+int(dig2)
    print(y,num[y])


for x in range(len(dig1)):
    num2 = int(dig1[x])+int(num2)

soma = num2+dig2
aux = str(soma)
print(num[0],num[1])
print(soma)

if (aux[len(aux)-1] == '0'):
    if(num[0] == '5' and num[1] == '1' or num[1] == '2' or num[1] == '3' or num[1] == '4' or num[1] == '5'):
        print("MASTERCARD")
    elif(num[0] == '3' and num[1] == '7' or num[0] == '3' and num[1] == '4'):
        print("AMEX")
    elif(num[0] == '4'):
        print("VISA")
    else:
        print("INVALID")

File: modulo 1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(void)
{

    char str[19], str2[19];
    long int num2 = 0, num3, result1 = 0, aux, aux2, result2 = 0, endresult = 0;
    long int num = get_long("Numero do cartão (apenas numeros)? \n");

    aux2 = num;
    do
    {
        // soma dos segundos digitos
        aux = result1;
        num2 = (num % 10);
        num3 = (num / 100);
        num = (num3);
        result1 = (aux + num2);
    }
    while (num > 0);

    num = aux2 / 10;
    do
    {
        // inicio da conversao dos 1 digitos para string
        num2 = (num % 10);
        num3 = (num / 100);
        num = (num3);
        num2 = (num2 * 2);
        sprintf(str, "%li", num2);
        strcat(str2, str);
    }
    while (num > 0);
    long int num4 = atol(str2); // conversao de str para long int

    do
    {
        // soma dos primeiros digitos
        aux = result2;
        num2 = (num4 % 10);
        num3 = (num4 / 10);
        num4 = (num3);
        result2 = (aux + num2);
    }
    while (num4 > 0);

    endresult = (result1 + result2); // soma final


    num = aux2;
    do
    {
        // verificando qual cartão é coletando os 2 primeiros numeros
        num3 = (num / 10);
        num = (num3);
    }
    while (num3 >= 99);

    sprintf(str, "%li", aux2); //conversao

    if (endresult % 10 == 0) //imprimindo qual cartão é
    {
        if (strlen(str) > 16 || strlen(str) < 13)
        {
            printf("INVALID\n");
        }
        else if (num3 / 10 == 4)
        {
            printf("VISA\n");
        }
        else if (num3 == 34 || num3 == 37)
        {
            printf("AMEX\n");
        }
        else if (num3 >= 51 && num3 <= 55)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }

}